iris-backend-service/src/main/java/iris/backend_service/alerts/zammad/ZammadTicketCreator.java

```diff
@@ -0,0 +1,81 @@
+package iris.backend_service.alerts.zammad;
+
+import iris.backend_service.alerts.Alert;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * @author Jens Kutzsche
+ */
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ZammadTicketCreator {
+
+	private final RestTemplate rest;
```

Comment:
RestTemplate is supposed to be deprecated and should be replaced by WebClient: https://www.baeldung.com/spring-5-webclient

Reply:
Why should RestTemplate be deprecated? RestTemplate is for synchronous calls and WebClient is for asynchronous/reactive. These are two different valid approaches. We haven't used a reactive style yet and I wouldn't want to start using it now, even though I have some appreciation for the style.

Reply:
In the Tutorial for RestTemplate there is a deprecation notice (https://www.baeldung.com/rest-template point 2).

Reply:
In the code, the class is not yet marked as deprecated, but I took the hint as an opportunity to take a look at WebClient and use this after all.

I previously assumed that WebClient was only for reactive programming, so this is not true. Thanks for the hint.